src/cephadm/cephadm

```diff
@@ -4608,6 +4609,526 @@ def command_install():
+        return read_file(HostFacts._dmi_path_list, "bios_version")
+
+    @property
+    def bios_date(self):
+        # type: () -> str
+        """Determine server BIOS date from  DMI data in sysfs"""
+        return read_file(HostFacts._dmi_path_list, "bios_date")
+
+    @property
+    def timestamp(self):
+        # type: () -> float
+        """Return the current time as Epoch seconds"""
+        return time.time()
+    
+    @property
+    def system_uptime(self):
+        # type: () -> float
+        """Return the system uptime (in secs)"""
+        raw_time = read_file(['/proc/uptime'])
+        up_secs, _ = raw_time.split()
+        return float(up_secs)
+
+    @property
+    def kernel_security(self):
+        # type: () -> Dict[str, str]
+        """Determine the security features enabled in the kernel - SELinux, AppArmor"""
+        def _fetch_selinux():
+            """Read the selinux config file to determine state"""
+            security = {}
+            for selinux_path in HostFacts._selinux_path_list:
+                if os.path.exists(selinux_path):
+                    selinux_config = read_file([selinux_path]).splitlines()
+                    security['type'] = 'SELinux' 
+                    for line in selinux_config:
+                        if line.strip().startswith('#'):
+                            continue
+                        k, v = line.split('=')
+                        security[k] = v
+                    if security['SELINUX'].lower() == "disabled":
+                        security['description'] = "SELinux: Disabled"
+                    else:
+                        security['description'] = "SELinux: Enabled({}, {})".format(security['SELINUX'], security['SELINUXTYPE'])
+                    return security
+
+        def _fetch_apparmor():
+            """Read the apparmor profiles directly, returning an overview of AppArmor status"""
+            security = {}
+            for apparmor_path in HostFacts._apparmor_path_list:
+                if os.path.exists(apparmor_path):
+                    security['type'] = "AppArmor"
+                    security['description'] = "AppArmor: Enabled"
+                    try:
+                        profiles = read_file(['/sys/kernel/security/apparmor/profiles'])
+                    except OSError:
+                        pass
+                    else:
+                        summary = {}  # type: Dict[str, int]
+                        for line in profiles.split('\n'):
+                            item, mode = line.split(' ')
+                            mode= mode.strip('()')
+                            if mode in summary:
+                                summary[mode] += 1
+                            else:
+                                summary[mode] = 0
+                        summary_str = ",".join(["{} {}".format(v, k) for k, v in summary.items()])
+                        security = {**security, **summary} # type: ignore
```

Comment:
```
2020-08-18T12:37:27.216 INFO:tasks.workunit.client.0.smithi187.stderr:+ sudo /tmp/tmp.aT7YfqeW4S/cephadm --image docker.io/ceph/daemon-base:latest-master-devel rm-cluster --fsid 00000000-0000-0000-0000-0000deadbeef --force
2020-08-18T12:37:27.245 INFO:tasks.workunit.client.0.smithi187.stderr:  File "/tmp/tmp.aT7YfqeW4S/cephadm", line 5091
2020-08-18T12:37:27.245 INFO:tasks.workunit.client.0.smithi187.stderr:    security = {**security, **summary} # type: ignore
2020-08-18T12:37:27.245 INFO:tasks.workunit.client.0.smithi187.stderr:  
```

https://pulpito.ceph.com/swagner-2020-08-18_11:50:15-rados:cephadm-wip-swagner3-testing-2020-08-18-1033-distro-basic-smithi/5356265/

Reply:
ping @pcuzner

Reply:
that's a py2 error- why do we care, or test for py2, when a) it's EOL and b) we must have py3 to deploy and run cephadm anyway?
@sebastian-philipp ???

Reply:
do you want to remove py2 from test_cephadm.sh? See https://github.com/ceph/ceph/pull/36571#pullrequestreview-468230744

Reply:
I see no reason to have py2 in there, if you agree I'll remove.

Reply:
that would be great!

Reply:
updated the #36571 PR